### PR TITLE
chore: Upgrade Cypress TypeScript recipes

### PR DESCRIPTION
**Pull request by NicholasBoll**
After more experimenting and feedback, `"types"` is a better way to import type definitions. This should "just work" and reduce conflict issues with project already using TypeScript for the application.

### CODE REVIEW

File: examples/preprocessors__typescript-browserify/cypress/integration/spec.ts [hunk 1]
```diff
@@ -1,7 +1,5 @@
 import { add } from '../support/add'
 
-declare function expect(any):any
```
Comment (NicholasBoll) [inline examples/preprocessors__typescript-browserify/cypress/integration/spec.ts]: I don't think this is needed anymore. The Cypress type definitions include `chai` and `mocha` definitions.

File: examples/preprocessors__typescript-webpack/cypress/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	"strict": true,
+    "baseUrl": "../../../node_modules", // Point this to the root `node_modules` of your project. Most likely `../node_modules`
+    "target": "es5",
+    "lib": ["es5", "dom"],
+    "types": ["cypress"],
+  },
+  "include": [
+    "**/*.ts"
+  ]
+}
```
Comment (NicholasBoll) [inline examples/preprocessors__typescript-webpack/cypress/tsconfig.json]: @bahmutov I'll leave it up to you if you want all the comments. I removed for aesthetics (to make it obvious what is important to change)

File: examples/preprocessors__typescript-browserify/cypress/integration/spec.ts (hunk 1 above)
Comment (bahmutov) [reply examples/preprocessors__typescript-browserify/cypress/integration/spec.ts]: good catch, should not be necessary anymore

File: examples/preprocessors__typescript-webpack/cypress/tsconfig.json (hunk 2 above)
Comment (bahmutov) [reply examples/preprocessors__typescript-webpack/cypress/tsconfig.json]: sure, not objections

File: examples/preprocessors__typescript-webpack/README.md [hunk 3]
```diff
@@ -19,26 +19,23 @@ The E2E tests should run in Cypress right away. There are few other commands con
...
 
 Because `cypress` is installed in the root folder of this repo, [tsconfig.json](tsconfig.json) specifies where to find its types:
 
 ```
 {
-  "include": [
-    "../../node_modules/cypress/index.d.ts",
-    "cypress/*/*.ts"
-  ]
+  "types": ["cypress"],
```
Comment (bahmutov) [inline examples/preprocessors__typescript-webpack/README.md]: interesting - this seems to be equivalent to `/// reference types="cypress"` trick

**Comment by bahmutov** [comment]
arghhh, record flag stops Cypress tests https://github.com/cypress-io/cypress/issues/1193

**Comment by bahmutov** [comment]
Trying this pull request locally, getting an error in the browserify example

```
$ npm run cypress:run

> cypress-example-typescript-browserify@1.0.0 cypress:run /Users/gleb/git/cypress-example-recipes/examples/preprocessors__typescript-browserify
> ../../node_modules/.bin/cypress run

It looks like this is your first time using Cypress: 2.1.0

 ✔  Verified Cypress! /Users/gleb/git/cypress-example-recipes/node_modules/cypress/dist/Cypress.app

Opening Cypress...

Started video recording: /Users/gleb/git/cypress-example-recipes/examples/preprocessors__typescript-browserify/cypress/videos/sedmv.mp4

  (Tests Starting)
Oops...we found an error preparing this test file:

  cypress/integration/spec.ts

The error was:

TypeScript error: cypress/integration/spec.ts(14,5): Error TS2304: Cannot find name 'expect'.


This occurred while Cypress was compiling and bundling your test code. This is usually caused by:

- A missing file or dependency
- A syntax error in the file or one of its dependencies

Fix the error in your code and re-run your tests.
```

File: examples/preprocessors__typescript-webpack/README.md (hunk 3 above)
Comment (NicholasBoll) [reply examples/preprocessors__typescript-webpack/README.md]: Yes. The added benefit over `"includes"` is to isolate the types being used in the Cypress directory. I've seen some people have issues with type conflicts if they happen to use namespace types (globals) in their application code like `chai` or `jquery`. Just having "cypress" will use the "cypress" types as the root and the Cypress type definitions will pull in dependencies.

**Comment by NicholasBoll** [comment]
I saw  the record flag and didn't know why it failed. Makes sense with the linked issue.

**Comment by NicholasBoll** [comment]
@bahmutov I'll look into that failure and fix it

**Comment by bahmutov** [comment]
implemented same thing in https://github.com/cypress-io/cypress-example-recipes/commit/33dab4d29a1a1295b8b9805582dae27cd74a31a5

**Pull request closed without merging**